test/NuGet.Core.Tests/NuGet.PackageManagement.Test/PackageRestoreManagerTests.cs

```diff
@@ -353,7 +368,7 @@ public async Task Test_PackageRestoreFailure_WithRaisedEvents()
                 nuGetPackageManager.PackageExistsInPackagesFolder(testPackage2).Should().BeFalse();
 
                 // Act
-                await packageRestoreManager.RestoreMissingPackagesInSolutionAsync(testSolutionManager.SolutionDirectory,
+                PackageRestoreResult result = await packageRestoreManager.RestoreMissingPackagesInSolutionAsync(testSolutionManager.SolutionDirectory,
```

Comment:
This isn't required, but I saw it returned something and prefer seeing that. I could use a discard. Other opinions?

Reply:
I never investigated why it's different in some projects, but some projects cause a compiler warning (or maybe it's just an info/underline?) when variables are not used.

But if you add a `result.Should().NotBeNull()` then it'll be used in the test, and available during debugging for inspection.

Reply:
Let me merge this and see how it helps the flakiness, and I'll do that in another PR.